RELEASE-NOTES.md

```diff
@@ -1,5 +1,17 @@
 # Wikibase DataModel release notes
 
+## Version 6.0.0 (dev)
+
+* Removed `Entity` class (deprecated since 1.0)
```

Comment:
I don't understand why we need to kill the class and refactor everything in a single step, in a patch that exclusively talks about new interfaces. We can easily introduce the interfaces and only remove the conflicting methods from Entity.